### ISSUE TITLE
Documentation not deployed

### ISSUE THREAD

**Issue by cserteGT3** [JuliaGizmos/Observables.jl#41]
I started to revisit the problems of deploying the documentation, when I realised, that it has been already done. Although it is still not working.
[Latest build](https://travis-ci.org/JuliaGizmos/Observables.jl/jobs/654656611#L308) shows a `Permission denied` error, and suggests that is may caused by a wrong `DOCUMENTER_KEY`.

Another problem could be, that every job tries to deploy the documentation. If all of them succeeds, there will be four (six?) identical copies of the same version?
Though I'm not sure that it is a real problem, the following travis script avoids this:
```
jobs:
   allow_failures:
     - julia: nightly
   include:
     - stage: Documentation
       julia: 1.0
       os: linux
       script:
         - julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd()));
                                               Pkg.instantiate()'
         - julia --project=docs/ docs/make.jl
       after_success: skip
```
I can open a PR for the second, but can't fix or test the first.

**Comment by briochemc** [comment]
I think this is a problem too. Current deployed docs are maxed out at v0.1.2

<img width="225" alt="Screen Shot 2020-10-09 at 4 19 17 pm" src="https://user-images.githubusercontent.com/4486578/95546058-4fa57300-0a4b-11eb-9f5e-34701961716b.png">

 when current latest version is v0.3.2:

<img width="1326" alt="Screen Shot 2020-10-09 at 4 21 19 pm" src="https://user-images.githubusercontent.com/4486578/95546115-8085a800-0a4b-11eb-87f4-a92ae4f76af9.png">

**Comment by cserteGT3** [comment]
[Latest build](https://travis-ci.org/github/JuliaGizmos/Observables.jl/jobs/733021891) shows that the `DOCUMENTER_KEY` variable doesn't set up correctly.

<details>
<summary>Copied the CI results here</summary>

```
Initialized empty Git repository in /private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/jl_a9XbyY/.git/

Warning: Permanently added 'github.com,140.82.112.3' (RSA) to the list of known hosts.

git@github.com: Permission denied (publickey).

fatal: Could not read from remote repository.

Please make sure you have the correct access rights

and the repository exists.

┌ Error: Git failed to fetch git@github.com:JuliaGizmos/Observables.jl.git

│ This can be caused by a DOCUMENTER_KEY variable that is not correctly set up.

│ Make sure that the environment variable is properly set up as a Base64-encoded string

│ of the SSH private key. You may need to re-generate the keys with DocumenterTools.

└ @ Documenter ~/.julia/packages/Documenter/pjwqp/src/Documenter.jl:546

┌ Error: Failed to push:

│   exception =

│    failed process: Process(`git fetch upstream`, ProcessExited(128)) [128]

│    

│    Stacktrace:

│     [1] pipeline_error at ./process.jl:525 [inlined]

│     [2] #run#565(::Bool, ::typeof(run), ::Cmd) at ./process.jl:440

│     [3] run at ./process.jl:438 [inlined]

│     [4] (::Documenter.var"#git_commands#25"{String,SubString{String},String,Array{Any,1},Bool,String,String})(::String) at /Users/travis/.julia/packages/Documenter/pjwqp/src/Documenter.jl:544

│     [5] (::Documenter.var"#24#30"{String})() at /Users/travis/.julia/packages/Documenter/pjwqp/src/Documenter.jl:649

│     [6] cd(::Documenter.var"#24#30"{String}, ::String) at ./file.jl:104

│     [7] #23 at /Users/travis/.julia/packages/Documenter/pjwqp/src/Documenter.jl:649 [inlined]

│     [8] withenv(::Documenter.var"#23#29"{String,String}, ::Pair{String,String}) at ./env.jl:161

│     [9] (::Documenter.var"#22#28"{String,String,SubString{String},String})(::String, ::IOStream) at /Users/travis/.julia/packages/Documenter/pjwqp/src/Documenter.jl:648

│     [10] mktemp(::Documenter.var"#22#28"{String,String,SubString{String},String}, ::String) at ./file.jl:611

│     [11] mktemp(::Function) at ./file.jl:609

│     [12] #git_push#19(::String, ::String, ::String, ::SubString{String}, ::String, ::Array{Any,1}, ::Bool, ::Documenter.Travis, ::String, ::Bool, ::typeof(Documenter.git_push), ::String, ::String, ::String) at /Users/travis/.julia/packages/Documenter/pjwqp/src/Documenter.jl:633

│     [13] #git_push at ./none:0 [inlined]

│     [14] #16 at /Users/travis/.julia/packages/Documenter/pjwqp/src/Documenter.jl:498 [inlined]

│     [15] #mktempdir#18(::String, ::typeof(mktempdir), ::Documenter.var"#16#18"{SubString{String},String,String,String,String,Array{Any,1},Bool,Documenter.Travis,String,String,String,Bool}, ::String) at ./file.jl:634

│     [16] mktempdir(::Function, ::String) at ./file.jl:632 (repeats 2 times)

│     [17] (::Documenter.var"#15#17"{String,String,String,Nothing,String,Array{Any,1},Bool,Documenter.Travis,String,String,String,Bool})() at /Users/travis/.julia/packages/Documenter/pjwqp/src/Documenter.jl:497

│     [18] cd(::Documenter.var"#15#17"{String,String,String,Nothing,String,Array{Any,1},Bool,Documenter.Travis,String,String,String,Bool}, ::String) at ./file.jl:104

│     [19] (::Documenter.var"#kw##deploydocs")(::NamedTuple{(:repo,),Tuple{String}}, ::typeof(deploydocs)) at /Users/travis/.julia/packages/Documenter/pjwqp/src/Documenter.jl:474

│     [20] top-level scope at /Users/travis/build/JuliaGizmos/Observables.jl/docs/make.jl:12

│     [21] include at ./boot.jl:328 [inlined]

│     [22] include_relative(::Module, ::String) at ./loading.jl:1105

│     [23] include(::Module, ::String) at ./Base.jl:31

│     [24] include(::String) at ./client.jl:424

│     [25] top-level scope at none:1

│     [26] eval(::Module, ::Any) at ./boot.jl:330

│     [27] exec_options(::Base.JLOptions) at ./client.jl:263

│     [28] _start() at ./client.jl:460

└ @ Documenter ~/.julia/packages/Documenter/pjwqp/src/Documenter.jl:654
```
</details>

@shashi or @SimonDanisch could you please look at this? Docs haven't been deployed since June 2018.

**Comment by travigd** [comment]
Should be fixed by ea311bbe9ba4bdcb633e8618a061e6678b728024 and 3d5ed567039b6892550569d1adfd548200d811f8 (see [the GH action log](https://github.com/JuliaGizmos/Observables.jl/runs/1243288845?check_suite_focus=true)).
